### PR TITLE
Release 3.5.0

### PR DESCRIPTION
**Pull request by chenliangde**

### CODE REVIEW

File: controllers/admin/NostoIndexController.php [hunk 1]
```diff
@@ -82,6 +82,14 @@ public function getIframeUrl(NostoSDKAccount $account)
         return null;
     }
```
Comment (supercid) [inline controllers/admin/NostoIndexController.php]: Missing docblock.

File: classes/services/NostoCustomerService.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * Fires the update for the given customer and account
+     * returns true if the update was successful
+     *
+     * @param Customer $customer
+     * @param NostoSDKAccount $account
+     * @return bool
+     */
+    private static function updateMarketingPermissionInCurrentContext(Customer $customer, NostoSDKAccount $account)
```
Comment (hannunosto) [inline classes/services/NostoCustomerService.php]: This prolly should be public? Not sure how it works if you call this method from lambda and it's private

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 All notable changes to this project will be documented in this file.
 This project adheres to Semantic Versioning(http://semver.org/).
 
+## 3.5.0
+- Update customer's marketing permission to nosto via api
```
Comment (hannunosto) [inline CHANGELOG.md]: nosto -> Nosto

File: classes/services/NostoCustomerService.php [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+ */
+class NostoCustomerService extends AbstractNostoService
+{
+    /**
+     * Customer updated event handler. It is called after Customer updated or created.
+     *
+     * @param Customer $customer event parameters
+     * @return bool
+     */
+    public function customerUpdated(Customer $customer)
```
Comment (hannunosto) [inline classes/services/NostoCustomerService.php]: Could be static (`isAllUpdated` must be changed to static as well)

File: classes/services/NostoCustomerService.php [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    /**
+     * Check if all store views that the customer belongs to
+     * were updated. Logs stores that failed to update.
+     *
+     * @param $updatedAccounts
+     * @return bool
+     */
+    private function isAllUpdated(array $updatedAccounts)
```
Comment (hannunosto) [inline classes/services/NostoCustomerService.php]: Could be static

File: classes/services/NostoCustomerService.php (hunk 4 above)
Comment (chenliangde) [reply classes/services/NostoCustomerService.php]: I think none static method is better here as this is a service.

File: controllers/admin/NostoIndexController.php (hunk 1 above)
Comment (chenliangde) [reply controllers/admin/NostoIndexController.php]: Fixed

**Pull request merged**